src/NinjaTools.FluentMockServer/FluentAPI/IFluentExpectationBuilder.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.ComponentModel;
+using System.Net;
+using System.Net.Http;
+using JetBrains.Annotations;
+using NinjaTools.FluentMockServer.Models.ValueTypes;
+
+namespace NinjaTools.FluentMockServer.FluentAPI
+{
+    /// <summary>
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public interface IBlankExpectation : IFluentInterface
+    {
+        IBlankExpectation WithBaseUrl(string url);
+        IWithRequest OnHandling(HttpMethod method = null, [CanBeNull] Action<IFluentHttpRequestBuilder> requestFactory = null);
+        IWithRequest OnHandlingAny([CanBeNull] HttpMethod method = null);
+    }
+    
+    /// <summary>
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public interface IFluentExpectationBuilder : IBlankExpectation , IWithRequest, IWithResponse
+    {
+      
+    }
+    /// <summary>
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public interface IWithResponse : IFluentInterface
+    {
+        IFluentExpectationBuilder And { get; }
+        IWithResponse WhichIsValidFor(int value, TimeUnit timeUnit = TimeUnit.Seconds);
+        MockServerSetup Setup();
+    }
+        
+    /// <summary>
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public interface IWithRequest : IFluentInterface
+    {
+        IWithResponse RespondWith(int statusCode, Action<IFluentHttpResponseBuilder> responseFactory = null);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the overloading mechanism instead of the optional parameters.](https://app.codacy.com/app/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4722826)